src/lib/components/ChatInput.svelte

```diff
@@ -1,76 +1,302 @@
+          // Get the image file from clipboard
+          const file = item.getAsFile();
+          if (!file) continue;
+
+          // Prevent default paste behavior
+          event.preventDefault();
+
+          // Show loading indicator
+          const uploadButton = document.querySelector('[aria-label="Upload image"]');
+          if (uploadButton) {
+            uploadButton.setAttribute('disabled', 'true');
+            uploadButton.setAttribute('title', 'Uploading...');
+          }
+
+          try {
+            // Upload the image using the user.uploadBlob method
+            const uploadResult = await user.uploadBlob(file);
+
+            if (!tiptap) {
+              console.warn("Tiptap editor not initialized");
+              return;
+            }
+
+            // Get the raw image URL without any processing
+            const imageUrl = uploadResult.url;
+
+            // Insert image into editor with the raw URL
+            tiptap.chain().focus().insertContent({
+              type: "image",
+              attrs: { src: imageUrl }
+            }).run();
+
+            // Update content state to ensure persistence
+            content = tiptap.getJSON();
+          } catch (error) {
+            console.error("Error uploading pasted image:", error);
+            toast.error("Failed to upload pasted image", { position: "bottom-right" });
+          } finally {
+            // Re-enable the upload button
+            if (uploadButton) {
+              uploadButton.removeAttribute('disabled');
+              uploadButton.setAttribute('title', 'Upload image');
+            }
+          }
+
+          // Only process the first image found
+          return;
+        }
+      }
+    };
+
+    node.addEventListener('paste', pasteHandler);
+
+    return {
+      destroy() {
+        node.removeEventListener('paste', pasteHandler);
+      }
+    };
+  }
+
+  async function handleFileProcess(event: Event) {
+    const input = event.target as HTMLInputElement;
+    if (!input.files?.length) return;
+
+    const file = input.files[0];
+    // Type guard to ensure file is defined
+    if (!file || !file.type.startsWith("image/")) return;
+
+    // Show loading indicator or disable the button while uploading
+    const uploadButton = document.querySelector('[aria-label="Upload image"]');
+    if (uploadButton) {
+      uploadButton.setAttribute('disabled', 'true');
+      uploadButton.setAttribute('title', 'Uploading...');
+    }
+
+    try {
+      // Upload the image using the user.uploadBlob method
+      const uploadResult = await user.uploadBlob(file);
+
+      if (!tiptap) {
+        console.warn("Tiptap editor not initialized");
+        return;
+      }
+
+      // Get the raw image URL without any processing
+      const imageUrl = uploadResult.url;
+
+      // Insert image into editor with the raw URL
+      tiptap.chain().focus().insertContent({
+        type: "image",
+        attrs: { src: imageUrl }
+      }).run();
```

Comment:
Do you think you could also treat the image preview more like Discord does? Give it a max height and width so it doesn't take up so much space in the ChatInput? Something like 300 x 300. You can use `content-fit` on it too so it scales down correctly.